Warn about sourcemap leakage (#10637)
getsentry/sentry-docs: docs/platforms/javascript/common/sourcemaps/uploading/webpack.mdx

sidebar_order: 1
---

<Note>

> 
> This guide assumes you are using a Sentry SDK on version `7.47.0` or higher.
> 
> If you are on an older version and you want to upload source maps we recommend upgrading your SDK to the newest version.

Reviewer: ```suggestion
This guide assumes you're using a Sentry **SDK version `7.47.0` or higher**. If you're on an older version and you want to upload source maps, we recommend upgrading your SDK to the newest version.
```

</Note>

You can use the Sentry webpack plugin to automatically create releases and upload source maps to Sentry when bundling your app.

## Automatic Setup

> The easiest way to configure uploading source maps with webpack is by using the Sentry Wizard:

Reviewer: ```suggestion
The easiest way to configure source map uploading with webpack is by using the Sentry Wizard:
```

<Include name="sourcemaps-wizard-instructions.mdx" />

> If you want to configure source maps upload with webpack manually, follow the steps below.

Reviewer: ```suggestion
If you'd rather configure source map uploading with webpack manually, follow the steps below.
```

<OrgAuthTokenNote />

```bash {filename:.env.sentry-build-plugin}
SENTRY_AUTH_TOKEN=___ORG_AUTH_TOKEN___
```

<PlatformSection notSupported={["javascript.gatsby"]}>

<SignInNote />

And the following webpack config:

```javascript {filename:webpack.config.js}
const { sentryWebpackPlugin } = require("@sentry/webpack-plugin");

module.exports = {
  // ... other config above ...

  devtool: "source-map", // Source map generation must be turned on
  plugins: [
    sentryWebpackPlugin({
      org: "___ORG_SLUG___",
      project: "___PROJECT_SLUG___",
      authToken: process.env.SENTRY_AUTH_TOKEN,
    }),
  ],
};
```

</PlatformSection>

<PlatformSection supported={["javascript.gatsby"]}>

Add the following to your `gatsby-node.js` file:

<SignInNote />

```javascript {filename:gatsby-node.js}
const { sentryWebpackPlugin } = require("@sentry/webpack-plugin");

exports.onCreateWebpackConfig = ({ actions }) => {
  actions.setWebpackConfig({
    plugins: [
      sentryWebpackPlugin({
        org: "___ORG_SLUG___",
        project: "___PROJECT_SLUG___",
        authToken: process.env.SENTRY_AUTH_TOKEN,
      }),
    ],
  });
};
```

</PlatformSection>

<Alert>

>   Generating sourcemaps may expose them to the public, potentially causing your
>   source code to be leaked. You can prevent your source maps from being publicly
>   accessible by configuring your server to deny access to `.js.map` files, or by
>   using the [Sentry Webpack Plugin's
>   `sourcemaps.filesToDeleteAfterUpload`](https://www.npmjs.com/package/@sentry/webpack-plugin#sourcemapsfilestodeleteafterupload)
>   option to delete source maps after they have been uploaded to Sentry.

Reviewer: ```suggestion
  Generating sourcemaps may expose them to the public, potentially causing your
  source code to be leaked. You can prevent this by configuring your server to deny access to `.js.map` files, or by
  using [Sentry Webpack Plugin's
  `sourcemaps.filesToDeleteAfterUpload`](https://www.npmjs.com/package/@sentry/webpack-plugin#sourcemapsfilestodeleteafterupload)
  option to delete source maps after they've been uploaded to Sentry.
```
